src/validators/fare_attributes.rs

```diff
@@ -0,0 +1,111 @@
+use crate::validators::issues::*;
+
+pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
+    let missing_price = gtfs
+        .fare_attributes
+        .values()
+        .filter(|fare_attributes| !has_price(*fare_attributes))
+        .map(|fare_attributes| make_issue(fare_attributes, IssueType::MissingPrice));
+    let invalid_currency = gtfs
+        .fare_attributes
+        .values()
+        .filter(|fare_attributes| !valid_currency(*fare_attributes))
+        .map(|fare_attributes| make_issue(fare_attributes, IssueType::InvalidCurrency));
+    let invalid_payment_method = gtfs
+        .fare_attributes
+        .values()
+        .filter(|fare_attributes| !valid_payment_method(*fare_attributes))
+        .map(|fare_attributes| make_issue(fare_attributes, IssueType::InvalidPaymentMethod));
+    let invalid_transfers = gtfs
+        .fare_attributes
+        .values()
+        .filter(|fare_attributes| !valid_transfers(*fare_attributes))
+        .map(|fare_attributes| make_issue(fare_attributes, IssueType::InvalidTransfers));
+    let invalid_duration = gtfs
+        .fare_attributes
+        .values()
+        .filter(|fare_attributes| !valid_duration(*fare_attributes))
+        .map(|fare_attributes| make_issue(fare_attributes, IssueType::InvalidTransferDuration));
+    missing_price
+        .chain(invalid_currency)
+        .chain(invalid_payment_method)
+        .chain(invalid_transfers)
+        .chain(invalid_duration)
+        .collect()
+}
+
+fn make_issue<T: gtfs_structures::Id>(o: &T, issue_type: IssueType) -> Issue {
+    Issue::new(Severity::Error, issue_type, o.id()).object_type(gtfs_structures::ObjectType::Fare)
+}
+
+fn has_price(fare_attributes: &gtfs_structures::FareAttribute) -> bool {
+    !fare_attributes.price.is_empty()
+}
+
+fn valid_currency(fare_attributes: &gtfs_structures::FareAttribute) -> bool {
+    iso4217::alpha3(&fare_attributes.currency).is_some()
+}
+
+fn valid_payment_method(fare_attributes: &gtfs_structures::FareAttribute) -> bool {
```

Comment:
it's a useless test as rust's enum cannot have any other value that the one defined.